File: templates/python/en/text/code/beeperLineBug1.py
# File: BeeperLineBug.py
# ------------------------------
# Uses a while loop to place a line of beepers.
# This program works for a world of any size.
# However, because each world requires one fewer
# moves than put_beepers it always misses a beeper.
from karel.stanfordkarel import *

# program starts at main
def main():
   # repeats until karel faces a wall
   while front_is_clear():
      # place a beeper on current square
      put_beeper()
      # move to the next square
      move()

File: templates/java/ja/text/code/beeperLineBug1.java
/*
 * File：Beeper行のバグ.java
 * -------------------
 * whileループを使用してbeeper秒の行をbeeperます。このプログラムは、あらゆる規模の世界で機能します。それぞれの世界が1つの少なく必要とするためしかし、 moveよりS putBeeper秒それは常にミスbeeper 。
 */
import stanford.カレル.*;
public class Beeper行のバグ extends カレル {
   public void run() {
   	  /* カレルが壁に面するまで繰り返す */
      while(frontIsClear()) {
      	 /* 現在の広場にbeeperを配置する */ 
         putBeeper();
      	 /* 次の広場へmove */
         move();
      }
   }
}